network/peer.go

```diff
@@ -53,6 +53,8 @@ type peer struct {
 
 	// unix time of the last message sent and received respectively
 	lastSent, lastReceived int64
+
+	lock sync.Mutex
```

Comment:
Can we document that only one peer lock should only be held at any given time by a thread of execution?